pkg/kubernetes/serviceaccount.go

```diff
@@ -90,3 +94,22 @@ func MaybeCreateServiceAccountOrUpdateMetadata(clientSet Interface, meta metav1.
 	logger.Info("updated serviceaccount %q", name)
 	return nil
 }
+
+// MaybeDeleteServiceAccount will only delete the serviceaccount if it exists
+func MaybeDeleteServiceAccount(clientSet Interface, meta metav1.ObjectMeta) error {
```

Comment:
You can't delete something that doesn't exist so I would just name this "DeleteServiceAccount()"

A better API if you want this to be reused for both when the caller needs it to fail when it didn't exist and when it shouldn't fail is to return a bool that conveys that.